core/src/main/java/com/exadel/etoolbox/querykit/core/models/query/QueryStatisticModel.java

```diff
@@ -0,0 +1,256 @@
+        } catch (RepositoryException e) {
+            return null;
+        }
+    }
+
+    private List<QueryStatisticModel.QueryInfo> getQueryInfo(MBeanServerConnection server, QueryManager queryManager) {
+        ObjectName queryStatMbean = getQueryStatMBean(server);
+        if (queryStatMbean == null || queryManager == null) {
+            return Collections.emptyList();
+        }
+        String jsonRaw;
+        List<QueryStatisticModel.QueryInfo> result = new ArrayList<>();
+
+        try {
+            jsonRaw = server.invoke(queryStatMbean, "asJson", null, null).toString();
+
+        } catch (Exception e) {
+            return Collections.emptyList();
+        }
+
+        JsonParser jsonParser = new JsonParser();
+        JsonArray queries = jsonParser.parse(jsonRaw).getAsJsonArray();
+
+        for(int i = 0; i < queries.size(); i++) {
+            JsonObject jsonObject = queries.get(i).getAsJsonObject();
+            String statement = getAsString(jsonObject, "query");
+            if (StringUtils.contains(statement, "oak-internal") || StringUtils.startsWithIgnoreCase(statement, "explain ")) {
+                continue;
+            }
+            String language = getAsString(jsonObject, "language");
+            String thread = getAsString(jsonObject, "lastThreadName");
+            String lastExecuted = getAsString(jsonObject, "lastExecutedMillis");
+            int executeCount = getAsInt(jsonObject, "executeCount");
+
+            QueryStatisticModel.QueryInfo queryInfo = new QueryStatisticModel.QueryInfo(queryManager);
+            queryInfo.setLanguage(language);
+            queryInfo.setStatement(statement);
+            queryInfo.setTheadName(thread);
+            queryInfo.setExecuteCount(executeCount);
+            queryInfo.setLastExecuted(lastExecuted);
+            result.add(queryInfo);
+        }
+        result.sort((a, b) -> Integer.compare(a.getExecuteCount(), b.getExecuteCount()) * -1);
+        return result;
+    }
+
+    private ObjectName getQueryStatMBean(MBeanServerConnection server) {
+        try {
+            Set<ObjectName> names = server.queryNames(new ObjectName("org.apache.jackrabbit.oak:type=QueryStats,*"), null);
+            return names.iterator().next();
+        } catch (IOException | MalformedObjectNameException | NoSuchElementException e) {
+            return null;
+        }
+    }
+
+    private List<String> getIndexes(ResourceResolver resourceResolver) {
+        Resource oakIndex = resourceResolver.getResource(OAK_INDEX_PREFIX);
+        if (oakIndex == null || !oakIndex.hasChildren()) {
+            return Collections.emptyList();
+        }
+        return StreamSupport.stream(oakIndex.getChildren().spliterator(), false)
+                .map(Resource::getPath)
+                .sorted()
+                .collect(Collectors.toList());
+    }
+
+    private String getIndexId(String value) {
+        if (value.contains("(") && value.contains(")")) {
```

Comment:
Please use Constants.OPENING_BRACKET, etc. (and please review other places when existing constants could be used

Reply:
Done